python/ray/serve/_private/application_state.py

```diff
@@ -557,7 +576,11 @@ def apply_deployment_info(
 
         return target_state_changed
 
-    def deploy_app(self, deployment_infos: Dict[str, DeploymentInfo]):
+    def deploy_app(
+        self,
+        deployment_infos: Dict[str, DeploymentInfo],
+        external_scaler_enabled: bool = False,
```

Comment:
Why have a default value for external_scaler_enabled? Since its guaranteed that the caller of this function will always pass a valid value for `external_scaler_enabled`, we should avoid having a default value here.